internal/commands/user/input_test.go

```diff
@@ -96,3 +99,191 @@ func TestProviderTypeDisplayUser(t *testing.T) {
 		})
 	}
 }
+
+func TestUserFilter(t *testing.T) {
+	t.Run("should create a user filter from inputs", func(t *testing.T) {
+		state := realm.UserStateDisabled
+		providerTypes := []string{"testProvider"}
+		users := []string{"testUser"}
+		inputs := multiUserInputs{
+			State:         state,
+			ProviderTypes: providerTypes,
+			Pending:       true,
+			Users:         users,
+		}
+		assert.Equal(t, realm.UserFilter{
+			IDs:       users,
+			State:     state,
+			Pending:   true,
+			Providers: realm.NewAuthProviderTypes(providerTypes...),
+		}, inputs.filter())
+	})
+}
+
+func TestResolveMultiUsersInputs(t *testing.T) {
+	testUsers := []realm.User{
+		{
+			ID:         "user-1",
+			Identities: []realm.UserIdentity{{ProviderType: realm.AuthProviderTypeAnonymous}},
+		},
+		{
+			ID:         "user-2",
+			Identities: []realm.UserIdentity{{ProviderType: realm.AuthProviderTypeUserPassword}},
+			Disabled:   true,
+			Data:       map[string]interface{}{"email": "user-2@test.com"},
+		},
+		{
+			ID:         "user-3",
+			Identities: []realm.UserIdentity{{ProviderType: realm.AuthProviderTypeUserPassword}},
+			Disabled:   true,
+			Data:       map[string]interface{}{"email": "user-3@test.com"},
+		},
+	}
+
+	t.Run("should resolve users", func(t *testing.T) {
+		for _, tc := range []struct {
+			description   string
+			inputs        multiUserInputs
+			users         []realm.User
+			expectedUsers []realm.User
+		}{
+			{
+				description:   "with no input set",
+				users:         testUsers,
+				expectedUsers: testUsers,
+			},
+			{
+				description:   "with all input set, except users",
```

Comment:
avoid punctuation in these descriptions, just go with `"with all input set except users"` here

---

also, i'm not sure what this test adds more than the one previous to this does?  for this `resolveUsers` function, id expect to see test cases for:

  - [x] when client errors, we error
  - [ ] when client returns no users, and inputs specified no user ids, we return an empty array
  - [x] when client returns no users, and inputs specified user ids, we error
  - [x] when a client returns users, we return those users

all other ones here should be able to be omitted, and you are probably left here with a test table of size 3 and one extra/error test.

Reply:
I cleaned up the test cases a bit. I _think_ the test cases present now account for the right subset of scenarios. I can add one for testing an error when selecting users. Looking at how to trigger that rn